src/resonate/scheduler.py

```diff
@@ -170,11 +206,28 @@ def __init__(
         self._durable_promise_storage = durable_promise_storage
         self._emphemeral_promise_memo: EphemeralPromiseMemo = {}
 
+        self._cmd_handlers = dict[type[Command], Callable[[list[Any]], list[Any]]]()
+        self._cmd_queues = dict[type[Command], deque[tuple[Command, Promise[Any]]]]()
+
         self._worker_thread = Thread(target=self._run, daemon=True)
         self._worker_thread.start()
 
-    def enqueue_cqe(self, sqe: _SQE[T], value: Result[T, Exception]) -> None:
-        self._completion_queue.put(_CQE[Any](sqe=sqe, fn_result=value))
+    def register_command_handler(
+        self,
+        cmd: type[C],
+        func: Callable[[list[C]], list[Any]],
```

Comment:
I think the command handler should be a function from `C -> Any` instead of `list[C] -> list[Any]`, similar to how `map` works.

Reply:
Probably not, lets think about this better.

Reply:
For example. Most GenAI api endpoints allow to calculate embeddings on multiple record at the same time, you can send a list of urls.

The resulting value of those are the embeddings of the input, in the same order